commands/survey.go

```diff
@@ -1,21 +0,0 @@
-package commands
-
-import survey "gopkg.in/AlecAivazis/survey.v1"
-
-// Survey is a command line survey.
-type Survey interface {
```

Comment:
we should have tests for survey. we already had some tests, why removing them within this commit?

Reply:
Because it was too complicated,

Again there was a new interface, new mock, mocks in the tests just to test `AskOne` method. We need to find a better way to test it. If we will mock everything that we can't tests we will end up in mock hell.

Also, those tests mix testing of validation (preRunE) and execution (runE) and good unit tests should test only one behavior/func at once (test one thing at a time).

Reply:
On [survey's README](https://github.com/AlecAivazis/survey#testing), the guy suggest to use (go-expect)[https://github.com/Netflix/go-expect] to catch stuff in stdout

Reply:
I don't approve removing the tests for survey. Having one more mock doesn't lead to a mock hell. I had a look on suvery's testing utilities yesterday but it seemed complicated to me, mocks are way simple. If you want to delete the tests anyway please put their replacement otherwise we will lose some meaningful tests.

Reply:
https://github.com/mesg-foundation/core/issues/622

Reply:
My motiviation to delete those tests was:
- mock hell what I mean by that if we swtich to passing interface to commands we will need: `daemon.Deamon`, `api.API`, `service.Service` (maybe) `CoreClient` mocks so it's already a lot. Adding one more mock seems good from short perspective, but in long run it's not so obious.

- so choosing mocking should be the last thing while testing. We alwasy need to try to find other solutions/design patterns that will free code of mocks. Why I think so. Look at go source code:
There is only one mock `net/dnsclient_unix_test.go:func` here. 

```
➜  moby git:(master) gogrep -i mock |grep struct |wc -l
36
```
So docker has 36 mocks and we probably haven't reached 2% of their codebase and we already have:
```
➜  core git:(feature/auto-remove-volume) ✗ gogrep -i mock |grep struct |wc -l
13
```

13 mocks.

- while mocking seems easy to go solution it is not in long run (compared to js where it's standard and it's super easy, but js are more flexible as it is dynamic programming language)

- survey suggested go-expect and vx10 for testing but vx10 is suspicious to me, but it's still better to use it as it doesn't require to introduce changes in the source code. 

- about using https://github.com/AlecAivazis/survey/blob/v1.7.0/confirm_test.go#L21 I was wrong. It can't be used.

So to summarize we should think of a better solution if possible, sleep with it, look from a different perspective, but introduce mocks because it easy should be avoided in general.

In was the same with `commands` package before `caputereStd` was created, there were no tests at all and in the meantime, we introduced them because we find a way to do so.

Reply:
Again, this is something that I just don't agree. There is no logical reason to remove a test if it's not replaced by an equivalent one. I'm not even talking about using the mocks here, it's not relevant.

Survey pkg doesn't provide a nice way of testing as I already checked so, it's a proper option to use mocks here. I don't mind adopting a **better way**. Could you please provide one? It seems that we're seeing things in different perspectives.  

> mock hell what I mean by that if we swtich to passing interface to commands we will need: daemon.Deamon, api.API, service.Service (maybe) CoreClient mocks so it's already a lot. Adding one more mock seems good from short perspective, but in long run it's not so obvious.

This is totally an another topic. If we want to reduce possibility for programming errors we should have tests about how _daemon_ and _coreapi_ used in the _commands/*_ pkgs. And I'd like doing that with unit tests. And I remember that you said it's not necessary to test *wrapper* funcs (the default implementation between these pkgs). And we don't have tests for them. I also [suggested](https://forum.mesg.com/t/automatize-manual-testing-for-core-true-integration-testing) to have at least some integration tests for reducing manual tests that we always required to do. Otherwise we don't have a way of catching bugs inside wrappers, the place where _commands_ pkg does IO which is very important.

I too prefer using the native testing utilities that can be provided by the pkg itself. But testing how _coreapi_ is used in commands pkg is easier by mocking the client side funcs of _coreapi_ as I experimented. We had the same problem before with the _go-application_ and _go-service_. We didn't use mocks there and implemented a fake _net.Conn_, which is a bit harder to maintain when compared to mocking because starting a fake gRPC server causes some async behavior that needed be synced otherwise it introduces bug into tests.

For daemon, we already have its interface so the best option is is to create mocks on top of that.

Reply:
> Survey pkg doesn't provide a nice way of testing as I already checked so, it's a proper option to use mocks here.

This is the point that it isn't good to use mocks. Again Survey pkg option for testing is still better then modifying code in order to introduce the Survey interface. Maybe there is an option to write Survery bridge. I don't know.

>  I don't mind adopting a better way. Could you please provide one?

I don't mind either, but as I said there is have no one at this moment, but shouldn't be the root cause of introducing mocks, because maybe there is another way that needs to be discovered.

For the rest of things is a different discussion, but in general, having too many tests as bad as having too little. And about the other things we can talk in different thread/PR